app/controllers/users_controller.rb

```diff
@@ -496,15 +496,29 @@ def is_local_username
     usernames -= groups
     usernames.each(&:downcase!)
 
-    cannot_see = []
+    cannot_see = {}
     here_count = nil
 
     topic_id = params[:topic_id]
     if topic_id.present? && topic = Topic.find_by(id: topic_id)
       usernames.each do |username|
-        if !Guardian.new(User.find_by_username(username)).can_see?(topic)
-          cannot_see.push(username)
+        user = User.find_by_username(username)
```

Comment:
Sorry not related to your change but I wonder if we can improve the performance here slightly. We trigger one DB query for each username to check here. I'm thinking of improving this by just fetching the users in a single DB query and then looping through that results instead.

Reply:
Just reading through the controller action in general but it doesn't seem like we have a limit on how many usernames we will check here. I think the means I can intentionally pass in a huge array of usernames which will trigger many db queries? Again not related to your change here but something I noticed in general while reading through the code here.